index.js

```diff
@@ -122,10 +122,22 @@ const plugin = fp(async function (app, opts) {
     schema = gateway.schema
     entityResolversFactory = gateway.entityResolversFactory
 
+    if (gateway.pollingInterval !== undefined) {
+      setInterval(async () => {
+        try {
+          fastifyGraphQl.schema = await gateway.refresh()
```

Comment:
This needs to be a bit more precise and tell us _if_ there are new schemas. We have a few internal caches for the parsed queries that must be invalidated if the schema change. In case of a change, I would also drop all outstanding websocket connections.

Reply:
I have replaced this line with

```js
fastifyGraphQl.replaceSchema(await gateway.refresh())
```

Reply:
This still need tests and logic to verify that when changing the schema the JIT cache is dropped and the websocket connections are closed, but only if the schema change and not every interval.